tests/rptest/services/redpanda.py

```diff
@@ -3286,11 +3305,42 @@ def cloud_storage_client(self):
         return self._cloud_storage_client
 
     def delete_bucket_from_si(self):
-        if self.si_settings.use_bucket_cleanup_policy:
-            self.logger.info(
-                f"Skipping deletion of bucket/container: {self.si_settings.cloud_storage_bucket}."
-                "Using a cleanup policy instead. Please delete it manually")
-            return
+        self.logger.info(
+            f"cloud_storage_cleanup_strategy = {self.si_settings.cloud_storage_cleanup_strategy}"
+        )
+
+        if self.si_settings.cloud_storage_cleanup_strategy == CloudStorageCleanupStrategy.ALWAYS_SMALL_BUCKETS_ONLY:
+            bucket_is_small = True
+            max_object_count = 3000
+
+            # See if the bucket is small enough
+            for i, m in enumerate(
+                    self.cloud_storage_client.list_objects(
+                        self.si_settings.cloud_storage_bucket)):
+                if i >= max_object_count:
+                    bucket_is_small = False
+                    break
+            if bucket_is_small:
+                self.logger.info(
+                    f"Always deleting bucket {self.si_settings.cloud_storage_bucket} (if it's small), even though we might use lifecycle policy also"
+                )
+            else:
+                self.logger.info(
+                    f"Bucket contains more than {max_object_count} objects, NOT cleaning."
+                )
+                return
+
+        elif self.si_settings.cloud_storage_cleanup_strategy == CloudStorageCleanupStrategy.IF_NOT_USING_LIFECYCLE_RULE:
+            if self.si_settings.use_bucket_cleanup_policy:
+                self.logger.info(
+                    f"Skipping deletion of bucket/container: {self.si_settings.cloud_storage_bucket}."
+                    "Using a cleanup policy instead. Please delete it manually"
```

Comment:
I am confused, there should be no scenario where you ever delete buckets manually after a test, right?

It's either cleaned up by the lifecycle rule/nuke (as before this change) or sync by the test, but never by manual deletion?

Reply:
This is an old existing log message. I'll fix it now.